conan/tools/apple/xcodedeps.py

```diff
@@ -223,6 +225,7 @@ def _content(self):
         all_deps = list(host_req.values()) + list(test_req.values())
         for dep in all_deps:
 
+            dep_cpp_info = self._conanfile.deps_cpp_info[dep.ref.name]
```

Comment:
``deps_cpp_info`` cannot be used anymore in new generators. Access is via ``self._conanfile.ref.name`` for conanfile and ``self.dependencies`` for dependencies.

Reply:
But it seems that `self.dependencies` does not contain the root path. How can I get that attribute?

Reply:
``self.dependencies["pkg"].package_folder``. The interface is described in https://docs.conan.io/en/latest/reference/conanfile/dependencies.html

Reply:
Thanks, changed the commit accordingly.